moz_kinto_publisher/main.py

```diff
@@ -254,85 +254,106 @@ def _get_attributes(self):
     def __str__(self):
         return "{{PEM: {} [h={} s={}]}}".format(self.filename, self.hash, self.size)
 
-    def verify(self, *, pemData=None):
-        # TODO: move to Intermediate which has self.certHash handy
-        localHash = hashlib.sha256(pemData.encode("utf-8")).hexdigest()
-        if localHash != self.hash:
-            return False
-        return True
+
+def allIn(keys, record):
+    return all(k in record for k in keys)
+
+
+def exactlyOneIn(keys, record):
+    return 1 == sum(k in record for k in keys)
 
 
 class Intermediate:
-    subject: str
-    kinto_id: str
-    whitelist: bool
-    crlite_enrolled: bool
-    pemAttachment: AttachedPem
     cert: x509.Certificate
-    certHash: str
-    subjectDN: bytes
+    crlite_enrolled: bool
     derHash: bytes
+    kinto_id: str
+    pemAttachment: AttachedPem
+    pemHash: str
     pubKeyHash: bytes
+    subject: str
+    subjectDN: bytes
+    whitelist: bool
 
     def __init__(self, debug=False, **kwargs):
-        self.pubKeyHash = base64.b64decode(
-            kwargs["pubKeyHash"], altchars="-_", validate=True
-        )  # sha256 of the SPKI
-        self.subject = kwargs["subject"]
-        self.whitelist = kwargs["whitelist"]
-
-        self.pemData = None
-        if "pem" in kwargs:
-            self.pemData = kwargs["pem"]
-
-        self.pemAttachment = None
-        if "attachment" in kwargs:
-            self.pemAttachment = AttachedPem(**kwargs["attachment"])
-
-        if "enrolled" in kwargs:
-            self.crlite_enrolled = kwargs["enrolled"]
-        elif "crlite_enrolled" in kwargs:
-            self.crlite_enrolled = kwargs["crlite_enrolled"]
-        else:
-            self.crlite_enrolled = False
-
+        self.derHash = None
         self.kinto_id = None
-        if "id" in kwargs:
-            self.kinto_id = kwargs["id"]
+        self.pemAttachment = None
+        self.pemData = None
+        self.pemHash = None
+        self.subjectDN = None
 
-        if len(self.pubKeyHash) < 26:
-            raise IntermediateRecordError(f"Invalid intermediate hash: {kwargs}")
+        parseError = IntermediateRecordError(f"Malformed record: {kwargs}")
+
+        # Local records have a "pem" field.
+        # RemoteSettings records have "attachment".
+        # TODO(jms): These should be versioned.
```

Comment:
Let's file an issue and reference it here.

Reply:
Opened Issue #180 and updated comment in source.